File: node_api/models/Patient.js
const mongoose = require("mongoose");
const Schema = mongoose.Schema;

const PatientSchema = new Schema({
    name: {
        type: String,
        required: true
    },
    email: {
        type: String,
        required: true
    },
    password: {
        type: String,
        required: true
    },
    phone: {
        type: String
    },
    avatar: {
        type: String,
        default: 'default avatar path here'
    },
    age: {
        type: Number,
        required: true
    },
    city: {
        type: String,
        required: true
    },
    address: {
        type: String
    },
    gender: {
        type: String,
        required: true
    },
    reports: [
        {
            report: {
                type: Schema.Types.ObjectId,
                ref: 'reports'
            },
            heading: {
                type: String
            },
            date: {
                type: String
            }
        }
    ]
});

module.exports = Patient = mongoose.model("patients", PatientSchema);

File: medic_flutter_app/lib/models/patient.dart
import 'package:flutter/cupertino.dart';

class Patient {
  final String name;
  final String userId;
  final String email;
  // final String phone;
  final String avatar;
  final int age;
  final List<dynamic> reports;
  final String address;
  final String city;
  final String gender;
  final String token;
  Patient({
    @required this.name,
    @required this.userId,
    @required this.email,
    // @required this.phone,
    @required this.avatar,
    @required this.reports,
    @required this.age,
    @required this.address,
    @required this.gender,
    @required this.city,
    this.token,
  });
}
